packages/desktop-client/src/components/reports/ReportOptions.tsx

```diff
@@ -44,3 +44,92 @@ const intervalOptions = [
 { value: 5, description: 'Yearly', name: 5,
 ];
 */
+
+let uncategorizedCategory = {
+  name: 'Uncategorized',
+  id: null,
+  uncat_id: '1',
+  hidden: 0,
+  offBudget: false,
+  transfer: false,
+  category: false,
+};
+let transferCategory = {
+  name: 'Transfers',
+  id: null,
+  uncat_id: '2',
+  hidden: 0,
+  transfer: true,
+  offBudget: false,
+  category: false,
+};
+let offBudgetCategory = {
+  name: 'Off Budget',
+  id: null,
+  uncat_id: '3',
+  hidden: 0,
+  offBudget: true,
+  transfer: false,
+  category: true,
+};
+
+let uncategouncatGrouprizedGroup = {
+  name: 'Uncategorized & Off Budget',
+  id: null,
+  hidden: 0,
+  categories: [uncategorizedCategory, transferCategory, offBudgetCategory],
+};
+
+export const categoryLists = (uncat, categories) => {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏what's `uncat`? Variable naming could be improved.

Also missing function argument types.

Reply:
varibale naming was all moved to #1996 

Argument types added to entire file.